### ISSUE TITLE
[round 12] "multiple queries" benchmark for lapis vs openresty doesn't make sense

### ISSUE THREAD

**Issue by taoeffect** [TechEmpower/FrameworkBenchmarks#2045]
Re: https://www.techempower.com/benchmarks/#section=data-r12&hw=peak&test=query&l=b95g

This shows Lapis as over twice as fast as OpenResty, but how can that make sense given that Lapis is a framework *on top of* OpenResty? It should add overhead, not remove it.

**Comment by methane** [comment]
You can see difference from source.

**Comment by methane** [comment]
While I don't know much about Lua and OpenResty, OpenResty implementation uses lua-resty-mysql directly and connect to DB for each request.
https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Lua/openresty/app.lua

On the other hand, lapis uses PostgreSQL and implement connection pool.

**Comment by taoeffect** [comment]
Ah, well if they're using different databases and even different methods of querying, it definitely doesn't make sense to put them in the same set of benchmarks like that then. At that point you're not comparing frameworks but something completely different.

**Comment by agentzh** [comment]
> While I don't know much about Lua and OpenResty, OpenResty implementation uses
> lua-resty-mysql directly and connect to DB for each request.
> https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Lua/openresty/app.lua

Oh, this piece of Lua code is horrible. It's a shame to use it for benchmark :)

**Comment by agentzh** [comment]
@methane BTW, connection pools are implemented inside OpenResty (via ngx_lua's cosocket API) and both lua-resty-mysql and pgmoon utilize that.

**Comment by methane** [comment]
@agentzh If so, you can make openresty benchmark to use connection pool easily.
Contribution is welcome.

I'm contributing maintain Go and Python frameworks.
Since I'm not so interested in Lua, I won't learn it only for improve this benchmark.
I hope you can do it.

**Comment by methane** [comment]
Most benchmarks are from contributors.
If you're interested in some benchmark, reading source and improve it is better than asking
"why XXX is slow?" without research by yourselves.

**Comment by agentzh** [comment]
@methane I used to create pull requests to this project but my PRs were ignored.

**Comment by nbrady-techempower** [comment]
@agentzh Sorry that you feel that way! We try to get to PRs as quickly as possible. I noticed you had two PRs with us, one of them merged and it seems as though the other was closed without warning and then committed in a separate PR without tagging you. Are there any others we may have missed? We take each PR seriously and it often takes time to give them their due dilligence! Hopefully this won't deter you from contributing in the future! We appreciate all the help the community gives!
